google/cloud/pubsub/internal/publisher_tracing_connection.cc

```diff
@@ -44,18 +44,20 @@ opentelemetry::nostd::shared_ptr<opentelemetry::trace::Span> StartPublishSpan(
   namespace sc = opentelemetry::trace::SemanticConventions;
   opentelemetry::trace::StartSpanOptions options;
   options.kind = opentelemetry::trace::SpanKind::kProducer;
-  auto span =
-      internal::GetTracer(internal::CurrentOptions())
-          ->StartSpan(topic + " send",
-                      {
-                          {sc::kMessagingSystem, "pubsub"},
-                          {sc::kMessagingDestinationName, topic},
-                          {sc::kMessagingDestinationTemplate, "topic"},
-                          {"messaging.pubsub.ordering_key", m.ordering_key()},
-                          {"messaging.message.total_size_bytes",
-                           static_cast<std::int64_t>(MessageSize(m))},
-                      },
-                      options);
+  auto span = internal::GetTracer(internal::CurrentOptions())
+                  ->StartSpan(topic + " send",
+                              {
+                                  {sc::kMessagingSystem, "pubsub"},
+                                  {sc::kMessagingDestinationName, topic},
+                                  {sc::kMessagingDestinationTemplate, "topic"},
+                                  {"messaging.message.total_size_bytes",
+                                   static_cast<std::int64_t>(MessageSize(m))},
+                              },
+                              options);
+
+  if (!m.ordering_key().empty()) {
+    span->SetAttribute("messaging.pubsub.ordering_key", m.ordering_key());
```

Comment:
FYI: If we set the attribute after the span is created, our customers can no longer use the ordering_key attribute to make sampling decisions. Is that ok?

Reply:
Ack. I think it should be ok, but it's a good call out.